app/assets/javascripts/discourse/tests/unit/services/screen-track-test.js

```diff
@@ -26,4 +28,18 @@ discourseModule("Unit | Service | screen-track", function () {
       "caches highest read post number for second topic"
     );
   });
+
+  test("appEvent topic:timings-sent is triggered after posting consolidated timings", async function (assert) {
+    const appEvents = AppEvents.create();
+    const tracker = ScreenTrack.create({
+      appEvents,
+      _consolidatedTimings: [""],
+    });
+
+    appEvents.on("topic:timings-sent", () => {
+      assert.ok(true);
+    });
```

Comment:
This was supposed to test the service injection. Instead it sidesteps dependency injection system completely 😉

Reply:
@cvx,

My original tests were:

```ruby
  test("ScreenTrack has appEvents", async function (assert) {
    const tracker = this.container.lookup("service:screen-track");
    assert.ok(tracker.appEvents);
  });

  test("appEvent topic:timings-sent is triggered", async function (assert) {
    assert.timeout(1000);

    const tracker = this.container.lookup("service:screen-track");
    const appEvents = this.container.lookup("service:app-events");

    const done = assert.async();

    appEvents.on("topic:timings-sent", () => {
      assert.ok(true);
      done();
    });

    await tracker.sendNextConsolidatedTiming();
  });
```  
 
But they don't run. The service is injected fine when the application is running but in the tests they throw this error:
```
stack: >
            Error: Assertion Failed: Attempting to lookup an injected property on an object without a container, ensure that the object was instantiated via a container.
                at assert (http://localhost:7357/assets/vendor.js:47117:15)
                at ScreenTrack.getInjection (http://localhost:7357/assets/vendor.js:27252:55)
                ...
```

It happens whenever we try to access `appEvents` inside `ScreenTrack`.

Do you know how to fix this?

I've tried solving this all afternoon an night yesterday but couldn't figure it out and then asked for help. That's when Nat changed the test

Reply:
Something is not working quite right with `discourseModule` - it's not setting up the 'owner' correctly, so injections aren't working as expected. 😭 

We (well... @cvx 😉) has been working on moving us to Ember's officially recommended testing patterns. If we do that here, things seem to work much better.

```javascript
import { module, test } from "qunit";
import { setupTest } from "ember-qunit";

module("Unit | Service | screen-track", function (hooks) {
  setupTest(hooks);

  // ...(existing test, `this.container` -> `this.owner`)

  test("appEvent topic:timings-sent is triggered after posting consolidated timings", async function (assert) {
    const tracker = this.owner.lookup("service:screen-track");
    const appEvents = this.owner.lookup("service:app-events");

    appEvents.on("topic:timings-sent", () => {
      assert.ok(true);
    });

    tracker.consolidateTimings({ 1: 10, 2: 5 }, 10, 1);
    await tracker.sendNextConsolidatedTiming();
  });
});
```

Reply:
`module` it is, then!

Regarding:

```js
appEvents.on("topic:timings-sent", () => {
  assert.ok(true);
});
```

it originally used `async.done()` - I think it was a good idea. `assert true` doesn't guarantee the assertion is ever called (and if it is, it can be executed when the next test is already running)